luarules/mission_api/actions.lua

```diff
@@ -99,6 +100,43 @@ end
 
 --============================================================--
 
+-- Win Condition
+
+--============================================================--
+
+local function getAllyTeamsHavingPlayers()
+	local allyTeamsHavingPlayers = {}
+	for _, playerID in pairs(Spring.GetPlayerList()) do
+		local _, _, spec, _, allyTeamID = Spring.GetPlayerInfo(playerID, false)
+		if not spec and not allyTeamsHavingPlayers[allyTeamID] then
+			allyTeamsHavingPlayers[allyTeamID] = allyTeamID
+		end
```

Comment:
Don't need to worry about reassigning, cost is negligible.
```suggestion
		if not spec then
			allyTeamsHavingPlayers[allyTeamID] = allyTeamID
		end
```